src/javascript/app/pages/user/set_currency.js

```diff
@@ -85,6 +85,10 @@ const SetCurrency = (() => {
                                         redirect_url = Url.urlFor('user/authenticate');
                                     }
                                 }
+                                // Do not redirect MF clients to cashier, because they need to set self exclusion before trading
+                                if (!redirect_url && !/^(malta)$/i.test(Client.get('landing_company_shortcode'))) {
```

Comment:
have you tested this? as per the code IMO this won't work, on line 82 we are deciding for financial clients (MF) send them to authenticate page, and here we want to say that send gaming (MLT) clients to self exclusion instead, but in the comment you are mentioning `MF` instead of `MLT` and also you are checking that shortcode is not equal to `malta` while it should be equal to `malta`